app/models/concerns/profile_searching.rb

```diff
@@ -35,8 +35,22 @@ module ProfileSearching
       where(external: external)
     }
     scope :has_test_results, lambda { |has_test_results = true|
-      test_results = TestResult.select(:profile_id).distinct
-      has_test_results && where(id: test_results) || where.not(id: test_results)
+      # the use of default scope is to bypass Pundit
+      # and avoid the situation with bind parameters
```

Comment:
Maybe we could keep the existing behavior so we don't break existing API usage and just define a new search that is a bit more clear about what it returns, like `has_policy_test_results` or similar.

Reply:
Changed to add a new filter method instead of changing the existing one.